### PR TITLE
Rename DemoDataGenerator classes

### PR DESCRIPTION
**Pull request by rsynek**

### CODE REVIEW

File: quarkus-facility-location/src/main/java/org/acme/facilitylocation/bootstrap/FacilityLocationDataGenerator.java [hunk 1]
```diff
@@ -26,11 +26,11 @@
 import io.quarkus.runtime.StartupEvent;
 
 @ApplicationScoped
-public class DemoDataGenerator {
+public class FacilityLocationDataGenerator {
```
Comment (ge0ffrey) [inline quarkus-facility-location/src/main/java/org/acme/facilitylocation/bootstrap/FacilityLocationDataGenerator.java:29]: I guess it should be FacilityLocationProblemDataGenerator as it doesn't generate just locations?

**Review by ge0ffrey — APPROVED** [review]
Good to merge if the FLP one is good too

**Comment by ge0ffrey** [comment]
When we sync the quarkus-school-timetabling example with the one in quarkus-quickstarts, we need to adjust the quarkus docs, because those say "DemoDataGenerator". Note that those quarkus docs are coming back into our docs too (and we'll just duplicat them there).

@emmurphy1 This might affect some of the already single-sourced docs too.

**Comment by triceo** [comment]
> When we sync the quarkus-school-timetabling example with the one in quarkus-quickstarts, we need to adjust the quarkus docs, because those say "DemoDataGenerator". Note that those quarkus docs are coming back into our docs too (and we'll just duplicat them there).

This is exactly why it makes little sense to rename things just to rename things.
The fix is worse than the original problem - not only do you have to change references in many places (and potentially miss some, actually *creating* incosistency), you also *destroy what people already remember*. Everyone who ever blogged or made any kind of note about this will have their content outdated when this is merged. 

The hypothetical benefits from consistency in this case are, in my opinion, miniscule - especially since *the quickstarts were already consistent*, you just didn't like the name.

I implore you - stop doing this, @ge0ffrey . It helps no one, and it possibly hurts many.

**Comment by ge0ffrey** [comment]
@triceo Some renames are worth their fallout. Some restructures are worth their fallout. Like splitting off optaplanner-quickstart (which was your proposal IIRC). So I won't just stop polishing such things, especially in quickstarts/examples which need to be crystal clear for the getting started experience. This one is arguable, I don't feel strongly about it - happy to drop this request. We didn't realize the impact all the way into the quarkus docs when we noticed the downside of having so many classes with the same name.

@rsynek Wdyt? is this rename still worth the effort? Or shall we drop it? your call.

**Comment by rsynek** [comment]
It's true that we might impact some blog posts (although in our blog posts we don't mention this class) and as nobody feels strongly about merging this PR, I am closing it.

If we want to keep every potential blog post compatible, shall we only fix bugs after any quickstart makes it into the `stable` branch?

**Comment by ge0ffrey** [comment]
We don't want to keep blog posts compatible. That is a non-goal.

We want to keep the docs compatible, including those in quarkus.io, which's master needs to be moved to optaplanner-docs anyway.

**Pull request closed without merging**